src/message-io.js

```diff
@@ -85,18 +85,30 @@ module.exports = class MessageIO extends EventEmitter {
     return this._packetSize;
   }
 
-  startTls(credentialsDetails) {
+  startTls(credentialsDetails, trustServerCertificate) {
     const credentials = tls.createSecureContext ? tls.createSecureContext(credentialsDetails) : crypto.createCredentials(credentialsDetails);
 
     this.securePair = tls.createSecurePair(credentials);
     this.tlsNegotiationComplete = false;
 
-    this.securePair.on('secure', () => {
-      const cipher = this.securePair.cleartext.getCipher();
-      this.debug.log('TLS negotiated (' + cipher.name + ', ' + cipher.version + ')');
-      this.emit('secure', this.securePair.cleartext);
-      this.encryptAllFutureTraffic();
-    });
+    this.securePair.on('secure', ((_this) => {
+      return function() {
+        const cipher = _this.securePair.cleartext.getCipher();
+
+        if (!trustServerCertificate) {
+          const verifyError = this.ssl.verifyError();
+          if (verifyError) {
+            _this.securePair.destroy();
+            _this.socket.destroy();
+            _this.emit('error', verifyError);
```

Comment:
The `error` event only throws if there is no listener attached to it, otherwise it will not throw. These semantics are not very clear. 😞